cpmpy/solvers/gcs.py

```diff
@@ -0,0 +1,413 @@
+            Post a primitive CPMpy constraint to the GCS solver API
+        """
+        if isinstance(cpm_expr, _BoolVarImpl):
+            # base case, just var or ~var
+            return self.gcs.post_or([self.solver_var(cpm_expr)])
+        elif isinstance(cpm_expr, BoolVal):
+            if not cpm_expr:
+                return self.gcs.post_or([])
+            else:
+                return
+        elif isinstance(cpm_expr, Operator):
+            # 'and'/n, 'or'/n, '->'/2
+            if cpm_expr.name == 'and':
+                return self.gcs.post_and(self.solver_vars(cpm_expr.args))
+            elif cpm_expr.name == 'or':
+                return self.gcs.post_or(self.solver_vars(cpm_expr.args))
+
+            # Reified constraint: BoolVar -> Boolexpr or BoolVar == Boolexpr
+            # LHS must be boolvar due to only_bv_reifies
+            elif cpm_expr.name == '->' or cpm_expr.name == '==':
+                fully_reify = (cpm_expr.name == '==')
+                assert(isinstance(cpm_expr.args[0], _BoolVarImpl))  
+                bool_lhs = cpm_expr.args[0]
+                reif_var = self.solver_var(bool_lhs)
+                bool_expr = cpm_expr.args[1]
+
+                # Just a plain implies or equals between two boolvars
+                if isinstance(bool_expr, _BoolVarImpl): # bv1 -> bv2
+                    return (self.gcs.post_equals if fully_reify else self.gcs.post_implies)\
+                        (*self.solver_vars([bool_lhs, bool_expr]))
+                if isinstance(bool_expr, Operator): # bv -> and(...), bv -> or(...)  bv -> [->]
+                    if bool_expr.name == 'and':
+                        return self.gcs.post_and_reif(self.solver_vars(bool_expr.args), reif_var, fully_reify)
+                    elif bool_expr.name == 'or':
+                        return self.gcs.post_or_reif(self.solver_vars(bool_expr.args), reif_var, fully_reify)
+                    elif bool_expr.name == '->':
+                        return self.gcs.post_implies_reif(self.solver_vars(bool_expr.args), reif_var, fully_reify)
+                    else:
+                        # Shouldn't happen if reify_rewrite worked?
+                        raise NotImplementedError("Not currently supported by Glasgow Constraint Solver API '{}' {}".format)
+                
+                # Reified Comparison
+                elif isinstance(bool_expr, Comparison):
+                    lhs = bool_expr.args[0]
+                    rhs = bool_expr.args[1]
+                    if bool_expr.name == '==':
+                        return self.gcs.post_equals_reif(*self.solver_vars([lhs, rhs]), reif_var, fully_reify)
+                    elif bool_expr.name == '<=':
+                        return self.gcs.post_less_equal_reif(*self.solver_vars([lhs, rhs]), reif_var, fully_reify)
+                    elif bool_expr.name == '<':
+                        return self.gcs.post_less_if(*self.solver_vars([lhs, rhs]), reif_var, fully_reify)
```

Comment:
the other functions are post_X_reif and this one is post_X_if?

also on line 324 you use post_or_if while on 292 you use post_or_reif